ada_feeding/ada_feeding/helpers.py

```diff
@@ -357,12 +360,15 @@ def get_moveit2_object(
         blackboard.register_key(moveit2_lock_blackboard_key, Access.WRITE)
         # TODO: Assess whether ReentrantCallbackGroup is necessary for MoveIt2.
         callback_group = ReentrantCallbackGroup()
+        tool_joints = get_tool_joints(blackboard.get(end_effector_tool_blackboard_key))
```

Comment:
Add appropriate error handling around the `blackboard.get` if the key doesn't exist, doesn't have the right permissions, etc.

Reply:
Alternatively, you can pass `blackboard` to `get_tool_joints` and then handle the `get` from there (i.e., if any of the error cases arise, just set the key to `none`). That could help with centralization